### PR TITLE
Remove the testable library (and some typescript clean up)

### PR DESCRIPTION
**Pull request by scottpage1317**
- Replaces any components from the testable library with their underlying components
- Replaces TestPathSegment with a `tidRoot` which is defined in Form.js to the parent elements testID if `hasTestableWrappers` is set or `Element[index]` if's not.
- Fixes some typescript issues in `InputComponent` - I've tried to be as careful as possible to not change the behaviour of the field

I've tested this using flows which contain all the elements we support on Android and iOS (I'll keep this a draft until I test on Windows too), but I'm aware that doesn't cover all of the components in RNFG (the Switch and Link components for example) so am keen to get any suggestions for other places to test

### CODE REVIEW

**Comment by scottpage1317** [comment]
@GuyChatfield this is the PR which I think should be tested on Windows - how did you say I could test it without a working windows environment?

**Review by GuyChatfield — APPROVED** [review]
looks good to me 👍

File: src/Form.js [hunk 1]
```diff
@@ -83,6 +72,10 @@ export class Form extends Component {
       this
     );
 
-    return <View style={this.props.style}>{wrappedChildren}</View>;
+    return (
+      <View testID={`Form`} style={this.props.style}>
```
Comment (colincarter) [inline src/Form.js:76]: This `testId={'Form'}` could be replaced with `testID="Form"`

File: src/lib/DatePickerComponent.android.js [hunk 2]
```diff
@@ -159,69 +158,69 @@ export class DatePickerComponent extends React.Component {
       : "";
 
     const timeValue = this.state.date || new Date();
-    const valueTestId = this.state.date
-      ? `Value[${this.state.date.getTime()}]`
-      : "Value[unknown]";
+
+    const valueTestId =
+      `Value/` + this.state.date
+        ? `Value/${this.state.date?.getTime()}`
```
Comment (colincarter) [inline src/lib/DatePickerComponent.android.js:164]: `'Value/' + this.state.date` as a string of length > 0 will always evaluate to truthy, so the "Unknown" clause will never be.  ditto the ones below

File: src/lib/Field.js [hunk 3]
```diff
@@ -17,16 +15,16 @@ export class Field extends React.Component {
 
     if (this.props.onPress) {
       return (
-        <TTouchableOpacity onPress={this.props.onPress} tid={"OnPress"}>
+        <TouchableOpacity onPress={this.props.onPress} testID={`TapTarget`}>
```
Comment (colincarter) [inline src/lib/Field.js:18]: See above for the testIDs

File: src/lib/PickerComponent.ios.js [hunk 4]
```diff
@@ -29,26 +25,27 @@ class RenderedSelector extends React.Component {
...
         selectedValue={this.state.value}
         onValueChange={this.handleValueChange.bind(this)}
         mode="dropdown"
       >
         {this.props.options.map(
           ({ value, label }, idx) => (
-            <TPickerItem
-              tid={`PickerItem[${idx}]`}
+            <Picker.Item
+                testID={`PickerItem/${idx}`}
```
Comment (colincarter) [inline src/lib/PickerComponent.ios.js:40]: Is there some formatting issues here with the extra indentation?  Is prettier running?

File: src/lib/TimePickerComponent.android.js [hunk 5]
```diff
@@ -69,41 +67,39 @@ export class TimePickerComponent extends React.Component {
     let placeholderComponent = this.props.placeholderComponent ? (
       this.props.placeholderComponent
     ) : (
-      <TText tid="Placeholder" style={this.props.placeholderStyle}>
+      <Text testID={`Label`} style={this.props.placeholderStyle}>
```
Comment (colincarter) [inline src/lib/TimePickerComponent.android.js:70]: `Text` hasn't been imported here

File: src/lib/DatePickerComponent.android.js [hunk 6]
```diff
@@ -2,11 +2,10 @@
 
 import React from "react";
 import PropTypes from "prop-types";
-let { View, StyleSheet, TextInput, Button } = require("react-native");
+let { View, Text } = require("react-native");
```
Comment (colincarter) [inline src/lib/DatePickerComponent.android.js:5]: Could you use `import {View, Text} from "react-native"` here?

File: src/lib/DatePickerComponent.windows.js [hunk 7]
```diff
@@ -3,11 +3,10 @@
 import React from "react";
 import PropTypes from "prop-types";
 
-let { View, StyleSheet, TextInput } = require("react-native");
+let { View, Text } = require("react-native");
```
Comment (colincarter) [inline src/lib/DatePickerComponent.windows.js:6]: See above

File: src/lib/LinkComponent.js [hunk 8]
```diff
@@ -3,9 +3,6 @@
 import React from "react";
 let { View, StyleSheet, Text, ViewPropTypes } = require("react-native");
```
Comment (colincarter) [inline src/lib/LinkComponent.js:4]: See above

**Review by colincarter — CHANGES_REQUESTED** [review]
Thanks for the improvements @scottpage1317 .  Few changes to look at.

**Review by colincarter — APPROVED** [review]
Thanks, looks good now.

**Pull request merged**